Fixed some typos and a docstring

demberto opened the pull request.

demberto committed:
Fixed some typos and a docstring

afq984 commented on cxxfilt/__init__.py:
```diff
@@ -20,8 +20,7 @@ class InternalError(Error):
 class InvalidName(Error):
-    '''Exception raised when:
-    mangled_name is not a valid name under the C++ ABI mangling rules.'''
+    '''When mangled name is not a valid name under the C++ ABI mangling rules.'''
```
Suggest changing it to `mangled_name is not a valid name under the C++ ABI mangling rules`

`mangled_name` is an attribute of the Exception.

demberto replied:
I think it should stay that way, because `mangled_name` has no context in the body of the exception. (It has a meaning in the `__init__` method below, so `mangled_name` cam be used there)

afq984 replied:
`mangled_name` is used pretty consistently in the code so I don't think it lost meaning, but I'm fine with changing that. I believe "When" should be removed though.

demberto replied:
Revamped check new version

afq984 reviewed (commented):
Thanks for the fix. Just a minor nit.

demberto committed:
Revamp

1. Added Google-style docstrings.
2. Exceptions are now how they should be.
3. Added `__all__` and `annotations`.
4. Replaced `.format()` with f-strings.

I don't have a Linux/MacOS machine so you should check all this.
Since I have changed the exception names, this should be a breaking release, you might rename them back to original names, but these names are more accurate.

demberto committed:
Merge pull request #1 from demberto/patch-1

Patch 1

demberto reviewed (commented):
Revamped version. Github confuses me like hell, I don't know if I did the right thing here

afq984 commented on cxxfilt/__init__.py:
```diff
@@ -1,117 +1,243 @@
+from __future__ import annotations
```
Is there a reason this ([PEP 563](https://www.python.org/dev/peps/pep-0563/)) is needed? 
Adding this apparently also breaks the EOL'd python 3.6

afq984 commented on cxxfilt/__init__.py:
```diff
@@ -1,117 +1,243 @@
+        return f"{self.mangled_name} is an invalid name"
 
-class CharP(ctypes.c_char_p):
-    pass
```
As I remember using its subclass actually cause some behavior change. Need to figure out what it was.

afq984 commented on cxxfilt/__init__.py:
```diff
@@ -1,117 +1,243 @@
-    pass
+# can be subclassed from FileNotFoundError?
+# see https://stackoverflow.com/a/36077407
+class LibraryNotFoundError(Error):
```
Even the error names are currently inconsistent I would avoid changing them as doing so might break existing users.

afq984 replied:
Also, we cannot subclass from `FileNotFoundError` (because it wasn't) as it changes observable behavior, such as

```
try:
    with open(input) as file:
        output = cxxfilt.demangle(file.read())
except FileNotFoundError:
    print(input, 'not found')
except cxxfilt.InvalidName:
    print(input, 'invalid')
else:
    print(output)
```

afq984 commented on cxxfilt/__init__.py:
```diff
@@ -1,117 +1,243 @@
+
+class InvalidNameError(Error):
+    """Raised when a mangled name is an invalid
+    name under the GNU C++ ABI mangling rules."""
```
> `GNU C++ ABI`

I'm not sure if the rules are GNU-defined. Can you provide some sources?

afq984 commented on cxxfilt/__init__.py:
```diff
@@ -1,117 +1,243 @@
-        return repr(self.mangled_name)
-
+    def __repr__(self) -> str:
+        return f"{self.mangled_name} is an invalid name"
```
I think this is redundant.

`{'a': 3}['b']` says `KeyError: 'b'` instead of `KeyError: key b is missing`

afq984 commented on cxxfilt/__init__.py:
```diff
@@ -1,117 +1,243 @@
         self._error = error
 
-    def demangleb(self, mangled_name: bytes, external_only: bool = True) -> bytes:
+    def demangleb(self, *_) -> bytes:
```
This will break existing code that calls using keyword arguments

afq984 commented on cxxfilt/__init__.py:
```diff
@@ -1,117 +1,243 @@
+            raise InvalidNameError(mangled_name)
         else:
-            raise InternalError('Unkwon status code: {}'.format(status.value))
+            raise InternalError(ret)
```
Do not move the status code handling into the `InternalError`. Having them all here makes it obvious how each return value is handled.
Also, the error codes aren't intended to be exposed to the user.

afq984 commented on cxxfilt/__init__.py:
```diff
@@ -1,117 +1,243 @@
 class Error(Exception):
+    """Base class for all cxxfilt exceptions."""
+
     pass
```
`pass` can be removed

afq984 commented on cxxfilt/__init__.py:
```diff
@@ -1,117 +1,243 @@
+
+    def __init__(self, choices: list[str]) -> None:
+        super().__init__()
+        self.choices = choices
```
Let's not expose the choices list from LibraryNotFound in this PR.

afq984 reviewed (changes requested):
Just take a look, some feedback:
* The code already uses single quotes. Don't change that (or make it its own PR).
* Exception names shouldn't be changed as existing code using them will break.
* Making the exceptions expose more info (such as `LibraryNotFoundError.choices` and `InternalError.ret_code`) should be in another PR. But I'm not with the idea now. I don't think users' code can really use that extra information.
* Still need to figure out the crash in CI

demberto committed:
Applied your recommendations

demberto committed:
Fixed some docstrings

demberto committed:
Merge pull request #2 from demberto/master

...

demberto reviewed (commented):
...
